lib/pages/account.dart

```diff
@@ -0,0 +1,442 @@
+import 'dart:convert';
+
+import 'package:capstone_app/main.dart';
+import 'package:flutter/material.dart';
+import 'package:capstone_app/services/auth.dart';
+import 'package:capstone_app/repositories/accountRepository.dart';
+import 'package:capstone_app/models/account.dart';
+import 'package:capstone_app/models/authentication.dart';
+import 'package:capstone_app/style/appTheme.dart';
+
+class AccountPage extends StatefulWidget {
+  AccountPage({Key key}) : super(key: key);
+
+  @override
+  _AccountPageState createState() => _AccountPageState();
+}
+
+class _AccountPageState extends State<AccountPage> {
+  GlobalKey<ScaffoldState> _scaffoldKey = GlobalKey<ScaffoldState>();
+
+  AccountRepository _accountRepo = AccountRepository.getInstance();
+  TextStyle _style;
+  Future<Account> accountInfo;
+  Account _account;
+  AuthArguments _args;
+
+  TextEditingController _firstNameController;
+  TextEditingController _lastNameController;
+  TextEditingController _phoneController;
+  GlobalKey<FormState> _formKey;
+
+  @override
+  initState() {
+    super.initState();
+//    _accountRepo.invalidateCurrentAccount();
+
+    _style = getAppTheme().primaryTextTheme.body1;
+    accountInfo = _accountRepo.getAccount();
+    _formKey = GlobalKey<FormState>();
+
+    accountInfo.then((Account account) async {
+      setState(() {
+        _account = account;
+      });
+
+  }).catchError((error) {
+      debugPrint('Error getting user info');
+
+      SnackBar snackBar = SnackBar(
+      content: Text('Could not connect to server'),
+    );
+
+    WidgetsBinding.instance.addPostFrameCallback((_) {
+    _scaffoldKey.currentState.showSnackBar(snackBar);
+    });
+
+  });
+
+
+
+  }
+
+  void submit() {
+
+
+    String email = _account.email;
+    String firstName = _firstNameController.text;
+    String lastName = _lastNameController.text;
+    String phone = _phoneController.text;
+    print("Submit function called");
+
+    _account.firstName = firstName;
+    _account.lastName = lastName;
+    _account.phoneNumber = phone;
+
+
+    if (_formKey.currentState.validate()) {
+      appAuth.updateAccount(email, firstName, lastName, phone).then((result) {
+        if (result != true) {
```

Comment:
If you want to ensure that changes made to account info are displayed after the update it made, you'll need to call setState somewhere